### PR TITLE
feat(mito2): Define basic structs for MitoEngine

### PR DESCRIPTION
**Pull request by evenyag**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?
This PR
- adds docs mod and class diagrams
- defines basic structs for MitoEngine

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/mito2/Cargo.toml [hunk 1]
```diff
@@ -5,3 +5,8 @@ edition.workspace = true
 license.workspace = true
 
 [dependencies]
+aquamarine = "0.3.2"
```
Comment (v0y4g3r) [inline src/mito2/Cargo.toml:8]:
```suggestion
aquamarine = "0.3"
```

**Review by v0y4g3r — APPROVED** [review]
LGTM, license header master

**Review by waynexia — APPROVED** [review]
LGTM, empty mod master

File: src/mito2/src/lib.rs [hunk 2]
```diff
@@ -12,4 +12,178 @@
...
+pub mod error;
+#[allow(dead_code)]
+mod region;
+#[allow(dead_code)]
+mod worker;
+
+#[cfg_attr(doc, aquamarine::aquamarine)]
+/// # Mito developer document
+///
+/// # Engine
```
Comment (waynexia) [inline src/mito2/src/lib.rs:32]:
```suggestion
/// ## Engine
```

File: src/mito2/src/lib.rs [hunk 3]
```diff
@@ -12,4 +12,178 @@
...
+/// MitoRegion o-- VersionControl
+/// MitoRegion -- ChunkReaderImpl
+/// WorkerGroup o-- RegionWorker
+/// RegionWorker o-- RegionMap
+/// RegionWorker -- RegionWorkerThread~LogStore~
+/// RegionWorkerThread~LogStore~ o-- RegionMap
+/// RegionWorkerThread~LogStore~ o-- Wal~LogStore~
+/// ```
+///
+/// # Metadata
```
Comment (waynexia) [inline src/mito2/src/lib.rs:90]:
```suggestion
/// ## Metadata
```

**Pull request merged**